File: dynamic_listings/lua/default/design_patterns/facade.lua
FirstService = {}
function FirstService:new(o)
-- Implementation of FirstService here...
end

SecondService = {}
function SecondService:new(o)
-- Implementation of SecondService here...
end

--[[
-- This class hides the complexities of using
-- FirstService and SecondService from the user
-- by "wrapping" them in a comfortable startAll
-- function
--]]
Facade = {}

function Facade:new(o)
    o = o or {
        service1 = FirstService.new({}),
        service2 = SecondService.new({})
    }
    setmetatable(o, self)
    self.__index = self
    return o
end

function Facade:startAll()
    --[[
    -- The facade starts all the services and does
    -- some status checking, this is hidden from the
    -- user.
    -- Returns true if all services started successfully
    -- false otherwise
    --]]
    local firstServiceStarted = self.service1.start()
    if (firstServiceStarted ~= true) then
        return false
    end
    local secondServiceStarted = self.service2.start()
    if (secondServiceStarted ~= true) then
        return false
    end
    -- Here everything started successfully
    return true;
end

File: dynamic_listings/javascript/default/design_patterns/facade.js
class FirstService{
    // Implementation here...
}

class SecondService{
    // Implementation here...
}

class Facade{
    /*
     * This class hides the complexities of using
     * FirstService and SecondService from the user
     * by "wrapping" them in a comfortable startAll
     * function
     */

    constructor(){
        this.service1 = new FirstService();
        this.service2 = new FirstService();
    }

    startAll(){
        /*
         * The facade starts all the services and does
         * some status checking, this is hidden from the
         * user.
         * Returns true if all services started successfully
         * false otherwise
         */
        let firstServiceStarted = this.service1.start();
        if (!firstServiceStarted){
            return false;
        }
        let secondServiceStarted = this.service2.start();
        if (!secondServiceStarted){
            return false;
        }
        // Here everything started successfully
        return true;
    }
}
